jetty-infinispan/infinispan-common/src/main/java/org/eclipse/jetty/session/infinispan/SessionDataMarshaller.java

```diff
@@ -49,6 +68,39 @@ public String getTypeName()
         return "org_eclipse_jetty_session_infinispan.InfinispanSessionData";
     }
 
+    @Override
+    public InfinispanSessionData readObject(ObjectInput input) throws IOException, ClassNotFoundException
+    {
+        if (serializationContext == null)
```

Comment:
I think you're going to need some synchronization - the lazy init of the serializationContext fields isn't thread safe, is it?

Reply:
You're right, better be sure